client/src/components/HistoryExport/ToRemoteFile.vue

```diff
@@ -0,0 +1,119 @@
+<template>
+    <div>
+        <b-alert v-if="errorMessage" show variant="danger" dismissible @dismissed="errorMessage = null">
+            {{ errorMessage }}
+            <JobError
+                v-if="jobError"
+                style="margin-top: 15px;"
+                header="History export job ended in error"
+                :job="jobError"
+            />
+        </b-alert>
+        <div v-if="waitingOnJob">
+            <loading-span message="Executing history export job, this will likely take a while." />
+        </div>
+        <div v-else-if="jobComplete">
+            <b-alert show variant="success" dismissible @dismissed="reset">
+                <h4>Done!</h4>
+                <p>History successfully exported.</p>
+            </b-alert>
+        </div>
+        <div v-else>
+            <b-form-group
+                id="directory"
+                label-for="directory"
+                description="Select a 'remote files' directory to export history archive to."
+                class="mt-3"
+            >
+                <files-input v-model="directory" mode="directory" :requireWritable="true" />
+            </b-form-group>
+            <b-form-group id="name" label-for="name" description="Give the exported file a name." class="mt-3">
+                <b-form-input id="name" v-model="name" placeholder="Name" required></b-form-input>
+            </b-form-group>
+            <b-row align-h="end">
+                <b-col
+                    ><b-button class="export-button" variant="primary" @click="doExport" :disabled="!canExport"
```

Comment:
I think it's a bit confusing that the button is just grey and you can't select any directories if no file sources are configured. I don't mind getting this in as a first version though, but maybe the `FilesInput ` component can make it clear that no sources are configured ?

Reply:
Revised this again, both import and export now load the available file source plugins. If there are not writable file sources, the tabs disappear and only the link option is available:

<img width="1212" alt="Screen Shot 2021-01-08 at 4 32 59 PM" src="https://user-images.githubusercontent.com/216771/104071503-4d33f000-51d7-11eb-8543-6c61e41129c5.png">

The import option likewise doesn't show that last radio button if there are not available file source plugins.